tests/unit/test_cli_flags.py

```diff
@@ -78,10 +81,81 @@ def test_prefer_param_value_to_user_config(self):
         flags = Flags(context, user_config)
         assert flags.USE_COLORS
 
-    def test_prefer_env_to_user_config(self, monkeypatch):
-        user_config = UserConfig(use_colors=False)
+    def test_prefer_env_to_user_config(self, monkeypatch, user_config):
+        user_config.use_colors = False
         monkeypatch.setenv("DBT_USE_COLORS", "True")
         context = self.make_dbt_context("run", ["run"])
 
         flags = Flags(context, user_config)
         assert flags.USE_COLORS
+
+    def test_mutually_exclusive_options_passed_separately(self):
+        """Assert options that are mutually exclusive can be passed separately without error"""
+        warn_error_context = self.make_dbt_context("run", ["--warn-error", "run"])
+
+        flags = Flags(warn_error_context)
+        assert flags.WARN_ERROR
+
+        warn_error_options_context = self.make_dbt_context(
+            "run", ["--warn-error-options", '{"include": "all"}', "run"]
+        )
+        flags = Flags(warn_error_options_context)
+        assert flags.WARN_ERROR_OPTIONS == WarnErrorOptions(include="all")
+
+    def test_mutually_exclusive_options_from_cli(self):
+        context = self.make_dbt_context(
+            "run", ["--warn-error", "--warn-error-options", '{"include": "all"}', "run"]
+        )
+
+        with pytest.raises(click.UsageError):
+            Flags(context)
+
+    @pytest.mark.parametrize("warn_error", [True, False])
+    def test_mutually_exclusive_options_from_user_config(self, warn_error, user_config):
+        user_config.warn_error = warn_error
+        context = self.make_dbt_context(
+            "run", ["--warn-error-options", '{"include": "all"}', "run"]
+        )
+
+        with pytest.raises(click.UsageError):
+            Flags(context, user_config)
+
+    @pytest.mark.parametrize("warn_error", ["True", "False"])
+    def test_mutually_exclusive_options_from_envvar(self, warn_error, monkeypatch):
+        monkeypatch.setenv("DBT_WARN_ERROR", warn_error)
+        monkeypatch.setenv("DBT_WARN_ERROR_OPTIONS", '{"include":"all"}')
+        context = self.make_dbt_context("run", ["run"])
+
+        with pytest.raises(click.UsageError):
+            Flags(context)
+
+    @pytest.mark.parametrize("warn_error", [True, False])
+    def test_mutually_exclusive_options_from_cli_and_user_config(self, warn_error, user_config):
+        user_config.warn_error = warn_error
+        context = self.make_dbt_context(
+            "run", ["--warn-error-options", '{"include": "all"}', "run"]
+        )
+
+        with pytest.raises(click.UsageError):
+            Flags(context, user_config)
+
+    @pytest.mark.parametrize("warn_error", ["True", "False"])
+    def test_mutually_exclusive_options_from_cli_and_envvar(self, warn_error, monkeypatch):
```

Comment:
We probably didn't have to have tests for some of these combos given that Click resolves the cli params and the env var params before we get to this stage in the code-- but I'll never complain about having more tests.